src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/EqualsAvoidNullCheck.java

```diff
@@ -174,30 +188,24 @@ public void finishTree(DetailAST ast) {
     }
 
     /**
-     * Determine whether SLIST begins static or non-static block and add it as
+     * Determine whether SLIST begins a block, determined by braces, and add it as
      * a frame in this case.
      * @param ast SLIST ast.
      */
     private void processSlist(DetailAST ast) {
-        final int parentType = ast.getParent().getType();
-        if (parentType == TokenTypes.SLIST
-                || parentType == TokenTypes.STATIC_INIT
-                || parentType == TokenTypes.INSTANCE_INIT) {
+        if (LEFT_CURLY.equals(ast.getText())) {
```

Comment:
`SLIST` is only a new frame if it has the curly. This is because switches have an SLIST with no brace in it and variables declared inside the switch are available to the entire switch, not just 1 case.